operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,345 @@
+        """
+
+        f = h5pyd.File("/nrel/wtk-us.h5", 'r')
+
+        # Setup date and time
+        dt = f["datetime"]
+        dt = pd.DataFrame({"datetime": dt[:]},index=range(0,dt.shape[0]))
+        dt['datetime'] = dt['datetime'].apply(dateutil.parser.parse)
+
+        project_idx = self.indicesForCoord(f)
+
+        print("y,x indices for project: \t\t {}".format(project_idx))
+        print("Coordinates of project: \t {}".format(self._lat_lon))
+        print("Coordinates of project: \t {}".format(f["coordinates"][project_idx[0]][project_idx[1]]))
+
+        # Get wind speed at 80m from the specified lat/lon
+        ws = f['windspeed_80m']
+        t_range = dt.loc[(dt.datetime >= start_date) & (dt.datetime < end_date)].index
+    
+        # Convert to dataframe
+        ws_tseries = ws[min(t_range):max(t_range)+1, project_idx[0], project_idx[1]]
+        ws_df=pd.DataFrame(index=dt.loc[t_range,'datetime'],data={'ws':ws_tseries})
+    
+        # Calculate diurnal profile of wind speed
+        ws_diurnal=ws_df.groupby(ws_df.index.hour).mean()
+
+        return ws_diurnal
+
+    def wtk_diurnal_plot(self):
+
+        """
+        This method plots the WTK diurnal plot alongisde the hourly power averages of the scada_df across all turbines
+        Args:
+        (None)
+        Returns: (None)
+        """
+
+        if self._id != 'None':
+            scada_sum = self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+            df_temp = scada_sum.copy()
+        else:
+            df_temp = self._df
+        df_diurnal = df_temp.groupby(df_temp[self._t].dt.hour)[self._w].mean()
+
+        ws_norm = self._ws_diurnal/self._ws_diurnal.mean()
+        df_norm = df_diurnal/df_diurnal.mean()
+    
+        plt.figure(figsize=(8,5))
+        plt.plot(ws_norm, label = 'WTK wind speed (UTC)')
+        plt.plot(df_norm, label = 'QC power')
+        plt.grid()
+        plt.xlabel('Hour of day')
+        plt.ylabel('Normalized values')
+        plt.title('WTK and QC Timezone Comparison')
+        plt.legend()
+        plt.show()
+
+    def corr_df_calc(self):
+        """
+        This method returns a correlation series that compares the current power data (with different shift thresholds) to wind speed data from the WTK with hourly resolution.
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+        if self._id != 'None':
+            scada_sum = self._df.groupby(self._df[self._t])[self._w].sum().to_frame()
+            df_diurnal = scada_sum.groupby(scada_sum[self._t].dt.hour)[self._w].mean()
+        else:
+            df_diurnal = self._df.groupby(self._df[self._t].dt.hour)[self._w].mean()
+
+        return_corr = np.empty((24))
+        for i in np.arange(24):
+            df_temp = df_diurnal.shift(i)
+        
+            if i != 0:
+                df_temp[np.arange(i)] = df_diurnal[-i:]
+
+            return_corr[i] = np.corrcoef(self._ws_diurnal['ws'], df_temp)[0,1]
```

Comment:
How about np.roll? I reckon lines 212-219 can be replaced with:

return_corr = np.empty((24))
for i in np.arange(24):
    return_corr[i] = np.corrcoef(self._ws_diurnal['ws'], np.roll(df_diurnal["ws"],i))[0,1]

Reply:
Applied

Reply:
I don't think this was actually applied in the code. But it's not a blocker.

Reply:
@jordanperr This change was originally made, and @moptis requested that the original order be preserved.